SmartDeviceLink/SDLProtocol.m

```diff
@@ -203,7 +203,7 @@ - (BOOL)sendRPC:(SDLRPCMessage *)message encrypted:(BOOL)encryption error:(NSErr
             // Build a binary header
             // Serialize the RPC data into an NSData
             SDLRPCPayload *rpcPayload = [[SDLRPCPayload alloc] init];
-            rpcPayload.functionID = [[[[SDLFunctionID alloc] init] getFunctionID:[message getFunctionName]] intValue];
+            rpcPayload.functionID = [[[SDLFunctionID sharedInstance] functionIdForName:[message functionNameForId]] intValue];
```

Comment:
seems we need a way to retrieve the id for a function name (as a response perhaps?) and the function name for the id (to send a request?). This is just a slight modification/improvement over how it's already been done. A full revamp can be done for 5.0.